crates/app/tests/helpers/tree.rs

```diff
@@ -79,3 +84,30 @@ pub async fn create_path(
         res.text().await.unwrap()
     );
 }
+
+pub async fn get_path(
+    cl: &reqwest::Client,
+    addr: &str,
+    repo: &RepositoryName,
+    tag: &TagName,
+    path: &TreePath,
+) -> (Vec<u8>, Mime, digest::ContentDigest) {
+    let url = format!("{}/{}/_tag/{}/tree{}", addr, repo, tag, path);
+
+    let res = cl.get(&url).send().await.unwrap();
+    assert_eq!(res.status(), StatusCode::OK);
+
+    let headers = res.headers().clone();
+
+    let length = headers.get(CONTENT_LENGTH).unwrap().clone();
+    let content_type = headers.get(CONTENT_TYPE).unwrap();
+    let content_type: Mime = Mime::from_str(content_type.to_str().unwrap()).unwrap();
+    let content_digest = headers.get("content-digest").unwrap().to_str().unwrap();
+    let resp_bytes = res.text().await.unwrap().clone().into_bytes();
```

Comment:
you should get a reader from the body and read it completely and then compare it's size with the `content-length` value.
`into_bytes` trims to `content-length` internally, whereas reader does not